src/chord/chord.rs

```diff
@@ -13,18 +13,43 @@ pub struct Chord {
     pub octave: u8,
     /// The intervals within the chord.
     pub intervals: Vec<Interval>,
-    /// The quiality of the chord; major, minor, diminished, etc.
+    /// The quality of the chord: major, minor, diminished, etc.
     pub quality: Quality,
-    /// The superscript number of the chord (3, 7, maj7, etc).
+    /// The superscript number of the chord: 3, 7, maj7, etc.
     pub number: Number,
+    /// The inversion of the chord: 0=root position, 1=first inversion, etc.
+    pub inversion: u8,
 }
 
 impl Chord {
     /// Create a new chord.
     pub fn new(root: PitchClass, quality: Quality, number: Number) -> Self {
+        Self::with_inversion(root, quality, number, 0)
+    }
+
+    /// Create a new chord with a given inversion.
+    pub fn with_inversion(
+        root: PitchClass,
+        quality: Quality,
+        number: Number,
+        inversion: u8,
+    ) -> Self {
+        let intervals = Self::chord_intervals(&quality, &number);
+        let inversion = inversion % (intervals.len() + 1) as u8;
+        Chord {
+            root,
+            octave: 4,
+            intervals,
+            quality,
+            number,
+            inversion,
+        }
+    }
+
+    pub fn chord_intervals(quality: &Quality, number: &Number) -> Vec<Interval> {
```

Comment:
Since this logic is encapsulated in a function now, there is no reason to pass the parameters by reference, pass them by value instead.